### PR TITLE
feat: allow setting of working directory 

### LINKED ISSUE #308: Local node requires sudo privileges
local node requires sudo privileges when ran with
```
hedera start
```

### PR DESCRIPTION
**Pull request by isavov**
**Description**:
This PR allows setting of the working directory where local node writes its ephemeral data which can be set using the `--workdir` flag, and has a default value dependant on the OS of the user
| OS      | Default Working Directory                    |
|---------|----------------------------------------------|
| MacOS   | `~/Library/Application Support/hedera-local` |
| Linux   | `~/.local/share/hedera-local`                |
| Windows | `%USERPROFILE%\AppData\Local\hedera-local`   |

**Related issue(s)**:

Fixes #308 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/utils/FileSystemUtils.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return join(homedir(), 'Library', 'Application Support', name);
+        }
+    
+        if (process.platform === 'win32') {
+            return join(process.env.LOCALAPPDATA || join(homedir(), 'AppData', 'Local'), name);
+        }
+        // else it's Linux
+        return join(process.env.XDG_DATA_HOME || join(homedir(), '.local', 'share'), name);
+    }
+}
```
Comment (georgi-l95) [inline src/utils/FileSystemUtils.ts:71]: nit: end of file

File: src/utils/FileSystemUtils.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        Object.entries(directories).forEach(([srcDir, destDir]) => {
+            FileSystemUtils.copyPath(srcDir, destDir);
+        });
+    }
+    
+    public static copyPath(srcPath: string, destPath: string): void {
+        if (!existsSync(srcPath)) return;
+    
+        if (lstatSync(srcPath).isDirectory()) {
+            FileSystemUtils.copyDirectoryContents(srcPath, destPath);
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply src/utils/FileSystemUtils.ts]: Fixed

File: src/utils/FileSystemUtils.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+export class FileSystemUtils{
+
+    public static copyPaths(directories: { [source: string]: string }): void {
+        Object.entries(directories).forEach(([srcDir, destDir]) => {
+            FileSystemUtils.copyPath(srcDir, destDir);
+        });
+    }
+    
+    public static copyPath(srcPath: string, destPath: string): void {
+        if (!existsSync(srcPath)) return;
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply src/utils/FileSystemUtils.ts]: fixed

**Review by stefan-stefanooov — APPROVED** [review]

File: src/state/InitState.ts [hunk 4]
```diff
@@ -65,19 +66,41 @@ export class InitState implements IState{
         const configurationData = new ConfigurationData().getSelectedConfigurationData(this.cliOptions.network);
         this.logger.info("Making sure that Docker is started and it's correct version...", this.stateName);
         // Check if docker is running and it's the correct version
-        const isCorrectDockerComposeVersion = await this.dockerService.isCorrectDockerComposeVersion();
```
Comment (konstantinabl) [inline src/state/InitState.ts:68]: Why is this deleted?

File: src/services/CLIService.ts [hunk 5]
```diff
@@ -267,7 +271,16 @@ export class CLIService implements IService{
...
-          });
+        });
+    }
+
+    private static workDirOption(yargs: Argv<{}>): void {
+        yargs.option('workdir', {
+            type: 'string',
+            describe: 'Path to the working directory for local node',
+            demandOption: false,
+            default: FileSystemUtils.getPlatformSpecificAppDataPath('hedera-local')
```
Comment (konstantinabl) [inline src/services/CLIService.ts:282]: Shouldnt this be hedera-local-node instead of hedera-local

**Review by konstantinabl — CHANGES_REQUESTED** [review]

File: src/state/InitState.ts (hunk 4 above)
Comment (isavov) [reply src/state/InitState.ts:68]: Sorry I was that the value of `isCorrectDockerComposeVersion` is never referenced and thought it was safe to remove. Putting it back in.

Comment (konstantinabl) [reply src/state/InitState.ts:68]: You can just remove the assignment to a variable, it is indeed not needed. Same goes to the line below

**Review by natanasow — APPROVED** [review]
LGTM, works on Ubuntu :rocket: :clap:

**Review by konstantinabl — APPROVED** [review]
